### PR TITLE
fix(registry): Fix the search filter in the payment registry

### LINKED ISSUE #2059: bug: Payments Registry "Exclude Reversed Payments" doesn't work
On the Cash Payments Registry, the "reversed" search parameter can either be:
 1. Include uniquely reversed records.
 2. Exclude all reversed records.

While (1) works perfectly, the (2) option doesn't seem to register at all.  This should be fixed as soon as possible since it is helpful to remove records that have been reversed from the analysis.  This bug does not  affect the Patient Invoice Registry.

### PR DESCRIPTION
**Pull request by DedrickEnc**
This P.R fixes the bug on the registry of payment when filtering by reversed equal zero
closes #2059

### CODE REVIEW

File: client/src/js/services/FilterService.js [hunk 1]
```diff
@@ -150,8 +150,8 @@ function FilterService(Store) {
 
     Object.keys(this._filterIndex).forEach(function (key) {
       var filter = this._filterIndex[key];
-
-      if (filter._value) {
+      
+      if (filter._value !== null && filter._value !== undefined && filter._value.length !== '') {
```
Comment (jniles) [inline client/src/js/services/FilterService.js:154]: I think you can just do `if (angular.isDefined(filter._value)) {`.

Comment (DedrickEnc) [reply client/src/js/services/FilterService.js:154]: ```angular.isDefined``` will condider ```null``` and empty string as value and will load every filter. This what I have got : 
filtres

Comment (jniles) [reply client/src/js/services/FilterService.js:154]: Fascinating.  I wonder how a user is able to enter `null`...

**Comment by jniles** [comment]
LGTM 

bors  r+

**Pull request merged**